nevergrad/benchmark/experiments.py

```diff
@@ -1181,5 +1217,45 @@ def bragg_structure(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
         for algo in default_optims if default_optims is not None else optims:
             yield Experiment(func, algo, int(budget), num_workers=1, seed=xpseed)
         for f in [func, func_nostruct, func_mix]:
-            for algo2 in recombinable:
-                yield Experiment(f, algo2, int(budget), num_workers=1, seed=xpseed)
+            for algo in recombinable:
+                yield Experiment(f, algo, int(budget), num_workers=1, seed=xpseed)
+
+
+# TODO : complete here for adversarial attacks.
+@registry.register
+def AdversarialAttack(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
+    class Normalize(nn.Module):
+        def __init__(self, mean, std):
+            super().__init__()
+            self.mean = torch.Tensor(mean)
+            self.std = torch.Tensor(std)
+
+        def forward(self, x):
+            return (x - self.mean.type_as(x)[None, :, None, None]) / self.std.type_as(x)[None, :, None, None]
+
+    class Resnet50(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.norm = Normalize(mean=[0.485, 0.456, 0.406],
+                                  std=[0.229, 0.224, 0.225])
+            self.model = resnet50(pretrained=True)
+
+        def forward(self, x):
+            return self.model(self.norm(x))
+
+    image_size = 224
+    data_folder = "/datasets01_101/imagenet_full_size/061417/val"
+    data_loader = torch.utils.data.DataLoader(
```

Comment:
If the path does not exist just stop (no yield command at all), so that this does not crash on CircleCI and the CI will be ok, whereas it will work on the dashboard.
For testing that everything works you can do 
python -m nevergrad.benchmark AdversarialAttack --seed=12 --repetitions=1 --plot

Reply:
Also, all this should be in the tesbed method of Adversarial, not in this file

Reply:
OK will correct that! but  why in testbed? I do not want my loader to load each  time, it is very costful no?

Reply:
> but why in testbed?

this way all adversarial example code stays in the corresponding package, and is reusable in other experiments without copy pasting everything, which is a **very** bad idea.

> I do not want my loader to load each time, it is very costful no?

How costful is it? That could be a good reason to avoid implementing this in testbed. Then again we could make it work without copying the loader, but that could require some work and I would be the one doing it. So if it's costy lets keep it that way for now.

Reply:
Costly like loading ImageNet every single time :p, so will keep it this way for now, and we'll see later :)

Reply:
actually I also want it to be modulable (ImageNet is one dataset, but I may want use any image), I want to specify an Image in my attack, not a dataset :)